### PR TITLE
Fext floatimage

### PR DESCRIPTION
**Pull request by rdiazgarHP**
Added interface to class FeatureExtractor such that we can feed FloatImage objects directly, besides image paths.

### CODE REVIEW

File: src/theia/sfm/feature_extractor.cc [hunk 1]
```diff
@@ -144,13 +150,41 @@ bool FeatureExtractor::ExtractFeatures(
 
     // Write the features to disk.
     CHECK(WriteKeypointsAndDescriptors(features_file, *keypoints, *descriptors))
-        << "Could not write features for image " << image_filename
-        << " from file " << features_file;
+    << "Could not write features for image " << image_filename
```
Comment (sweeneychris) [inline src/theia/sfm/feature_extractor.cc]: Indentation

File: src/theia/sfm/feature_extractor.cc [hunk 2]
```diff
@@ -102,34 +127,15 @@ bool FeatureExtractor::ExtractFeatures(
...
-                                                         descriptors)) {
+  if (!ExtractFeaturesFromImage(*image, keypoints, descriptors)) {
     LOG(ERROR) << "Could not extract descriptors in image " << filename;
     return false;
   }
-
-  if (keypoints->size() > options_.max_num_features) {
-    keypoints->resize(options_.max_num_features);
-    descriptors->resize(options_.max_num_features);
+  else {
```
Comment (sweeneychris) [inline src/theia/sfm/feature_extractor.cc]: In theia, we use brackets on the same line for if/else like this:

if (something) {
  DoTheThing();
} else {
  DoTheOtherThing();
}

Github isn't showing it in the comments, but we also use a 2-space indentation.

File: src/theia/sfm/feature_extractor.h [hunk 3]
```diff
@@ -88,6 +94,11 @@ class FeatureExtractor {
                        std::vector<Keypoint>* keypoints,
                        std::vector<Eigen::VectorXf>* descriptors);
 
+  // Extracts the features from a FloatImage
+  bool ExtractFeaturesFromImage(const FloatImage& image,
+                       std::vector<Keypoint>* keypoints,
```
Comment (sweeneychris) [inline src/theia/sfm/feature_extractor.h]: Indentation

**Comment by rdiazgarHP** [comment]
My bad. Please let me know if you spot some other issues.

**Comment by sweeneychris** [comment]
Is it possible to do a git squash these 3 commits into a single commit? that will make it easier for me to add the entire change in my review system Gerrit. Feel free to open a new PR with the single squashed commit if that is easier

**Comment by rdiazgarHP** [comment]
Squashed the 3 commits into one. Let me know if you need me to review something else.

**Pull request merged**